roles/node_exporter/templates/node_exporter.service.j2

```diff
@@ -16,7 +16,7 @@ ExecStart={{ node_exporter_binary_install_dir }}/node_exporter \
 {%     set name, options = (collector.items()|list)[0] -%}
     --collector.{{ name }} \
 {%     for k,v in options|dictsort %}
-    --collector.{{ name }}.{{ k }}={{ v | quote }} \
+    {{ ( '--collector.' + name + '.' + k + '=' + v  ) | quote }} \
```

Comment:
```suggestion
--collector.{{ name }}.{{ k }}={{ v | to_json }}
```

The issue is with the single quotes that the quote filter adds. The to_json filter adds double quotes to the value which is all that's needed.
See here: https://github.com/cloudalchemy/ansible-node-exporter/issues/176#issuecomment-742434170

Reply:
Interesting!
I traced the change down to this commit: https://github.com/cloudalchemy/ansible-node-exporter/pull/144
But it seems like everyone else except the author of that PR has issues with the quotes, for the sake of argument, could you try the exact syntax they are using? `=^(ram|loop|fd)\d+$`

Rather than adding quotes around the whole string, I would favor dropping the quotes altogether if there's no use cases for them.

Perhaps @SuperQ can shed some light into how this regex string should be formatted, since I'm not as familiar with the inner workings of node_exporter.

Reply:
Yes - seems like the quotes are unnecessary in my testing

This is in my template:

```yaml
--collector.{{ name }}.{{ k }}={{ v }} \
```

I added a couple of similar regex values to collector excludes:
```yaml
node_exporter_enabled_collectors:
- filesystem:
      fs-types-exclude: ^(nf|autof)\w+$
- diskstats:
      device-exclude: "^(ram|fd)\\d+$"
```
 
This is what I ended up with in the systemd unit config:
```
ExecStart=/usr/local/bin/node_exporter \
--collector.filesystem \
    --collector.filesystem.fs-types-exclude=^(nf|autof)\w+$ \
--collector.diskstats \
    --collector.diskstats.device-exclude=^(ram|fd)\d+$ \
```
Results - while not excluding loop devices, but excluding autofs filesystem types:
```
curl -s localhost:9100/metrics | grep -v ^# | grep -e 'disk_file.*loop0'
node_disk_filesystem_info{device="loop0",type="squashfs",usage="filesystem",uuid="",version="1024.0"} 1 

curl -s localhost:9100/metrics | grep -v ^# | grep -e 'filesys.*autofs'
~ $
curl -s localhost:9100/metrics | grep -v ^# | grep -e 'filesystem_files_free.*tmpfs'
node_filesystem_files_free{device="tmpfs",fstype="tmpfs",mountpoint="/run"} 4.109862e+06
node_filesystem_files_free{device="tmpfs",fstype="tmpfs",mountpoint="/run/snapd/ns"} 4.109862e+06
node_filesystem_files_free{device="tmpfs",fstype="tmpfs",mountpoint="/tmp"} 4.111084e+06
```
```suggestion
    --collector.{{ name }}.{{ k }}={{ v }} \
```

Reply:
Ok that looks promising!
I wonder if anything changed in node_exporter, awaiting feedback from @SuperQ before merging.

Reply:
Yes, the original addition of quotes is broken.

Perhaps we should pull in this patch: https://github.com/cloudalchemy/ansible-node-exporter/pull/245?